src/TypeSpec.Extension/Emitter.Csharp/test/Unit/usage.test.ts

```diff
@@ -253,4 +253,126 @@ describe("Test getUsages", () => {
         const usages = getUsages(sdkContext, services[0].operations);
         assert(usages.roundTrips.includes("Foo"));
     });
+
+    it("Test the usage of body polymorphism type in azure core resource operation.", async () => {
+        const program = await typeSpecCompile(
```

Comment:
Could we also add a test case that is using the enum-typed discriminator since this is one of the issue your PR trying to fix?

Reply:
yes. I will add test for polymorphism scenario UT. Thanks @ArcturusZhang

Reply:
@ArcturusZhang  Add discriminator Unit Test in Emitter.